### ISSUE TITLE
"200611031" internal error 

### ISSUE THREAD

**Issue by eugeneloza** [eugeneloza/decoherence#171]
Sometimes this one shows up during compilation... Either in Castle Game Engine or in one of Decoherence units.
Only info available is here http://free-pascal-general.1045716.n5.nabble.com/Internal-error-200611031-td5721646.html#a5721647

**Comment by michaliskambi** [comment]
I also get it sometimes. But never in case of a "clean compile". So cleaning + recompiling is always a working workaround (the build tool will even do this automatically.)

There's something in CGE dependencies (lower or at the same level as X3DNodes unit) that causes FPC to crash sometimes, but I was not able to reproduce it yet... X3DNodes is a huuge unit (worth splitting into many other units, but it's not easy). If you manage to reproduce it with a simple testcase, and submit to FPC team, I would be immensely grateful. But the root cause may be anywhere inside X3DNodes unit or lower, which includes > 50% CGE code...

In my experience it does not happen when editing code outside of the CGE, only when modifying the CGE code itself. So I do not treat it as "critical" now, it's just a nuisance for developers of CGE, not users of CGE. Are you able to reproduce it when editing only Decoherence code, not touching CGE?

**Comment by eugeneloza** [comment]
Yes, of course. I've left it as a note, and `Build and Clean` works for me perfectly. And it points exactly to X3DNodes unit near fonts definitions. I've experienced this issue twice during the last week. And of course it's not critical (and while it's a FPC bug, there's not much we can do here, only guess a small reproducible example).

**Comment by eugeneloza** [comment]
> Are you able to reproduce it when editing only Decoherence code, not touching CGE?

Not sure. I've ran into this error 2 or 3 times lately. 1 is certain in `X3DNodes`, another one - I don't remember exactly, but afair it was also `X3DNodes` (however, I don't remember if I changed anything in CGE then). This second (or third) time might have pointed to one of my units, but I was playing with project options at that time and did not pay enough attention, just rebuilt the project clean.
